extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,343 @@
+						this._controllerModel.refresh()
+					]);
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.refreshFailed(error));
+				}
+				finally {
+					refreshButton.enabled = true;
+				}
+			}));
+
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: refreshButton, toolbarSeparatorAfter: true }
+		]).component();
+	}
+
+	private createPodList(): string[] {
+		const podStatus = this._postgresModel.config?.status.podsStatus;
+		let podNames: string[] = [];
+
+		podStatus?.forEach(p => {
+			let podHealthModels: PodHealthModel[] = [];
+			let indexes: number[] = [];
+
+
+			p.conditions.forEach(c => {
+				const condtionContainer = this.modelView.modelBuilder.flexContainer().withProps({
+					CSSStyles: { 'alignItems': 'center', 'height': '15px' }
+				}).component();
+
+				if (c.status === 'False') {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.fail,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+						value: c.message,
+					}).component());
+				} else {
+					const imageComponent = this.modelView.modelBuilder.image().withProps({
+						iconPath: IconPathHelper.success,
+						width: iconSize,
+						height: iconSize,
+						iconHeight: '15px',
+						iconWidth: '15px'
+					}).component();
+					condtionContainer.addItem(imageComponent, { CSSStyles: { 'margin-right': '0px' } });
+
+					if (c.type === 'Initialized') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podInitialized,
+						}).component());
+					} else if (c.type === 'Ready') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podReady,
+						}).component());
+					} else if (c.type === 'ContainersReady') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.containerReady,
+						}).component());
+					} else if (c.type === 'PodScheduled') {
+						condtionContainer.addItem(this.modelView.modelBuilder.text().withProps({
+							value: loc.podScheduled,
+						}).component());
+					}
+				}
+
+				indexes.push(this.podsData.length);
+				this.podsData.push({
+					condition: c.type,
+					details: condtionContainer,
+					lastUpdate: c.lastTransitionTime
+				});
+			});
+
+			if (p.role.toUpperCase() !== loc.coordinator.toUpperCase()) {
+				podNames.push(p.name);
+			} else {
+				this.coordinatorData = podHealthModels;
+				this.coordinatorPodName = p.name;
+				podNames.unshift(p.name);
+			}
+			this.podConditionsTableIndexes.set(p.name, indexes);
+		});
+
+		this.podConditionsTable.data = this.podsData.map(p => [p.condition, p.details, p.lastUpdate]);
```

Comment:
To make sure I'm reading correctly - Does the table contain conditions for all pods, and get filtered for the particular pod selected in the dropdown?

Reply:
Yes. I thought this was better than resetting the table data each time a new pod was selected.